### PR TITLE
Add duration parameter to InitDiabloMsg()

### PR DESCRIPTION
**Pull request by kphoenix137**
Allows each DiabloMsg to have a set duration. Modified the `Game Saved` duration to 1 second (from default 3.5 seconds) as proof of concept and to quell a complaint. I'd like to make another PR to adjust various other messages that either last too long or not long enough (i.e. Debug Help menu is too short, various debug command messages last too long)

https://github.com/diasurgical/devilutionX/pull/6506

### CODE REVIEW

File: Source/error.cpp [hunk 1]
```diff
@@ -20,19 +20,24 @@ namespace devilution {
 
 namespace {
 
-std::deque<std::string> DiabloMessages;
+struct MessageEntry {
+	std::string text;
+	uint32_t duration;  // Duration in milliseconds
+	uint32_t startTime; // Time when the message started displaying
```
Comment (StephenCWills) [inline Source/error.cpp:26]: What is this for? It doesn't seem like it's used anywhere.

**Review by glebm — APPROVED** [review]

File: Source/error.cpp [hunk 2]
```diff
@@ -20,19 +20,23 @@ namespace devilution {
 
 namespace {
 
-std::deque<std::string> DiabloMessages;
+struct MessageEntry {
+	std::string text;
+	uint32_t duration;  // Duration in milliseconds
```
Comment (glebm) [inline Source/error.cpp:25]: Please fix clang-format issue here

Comment (AJenbo) [reply Source/error.cpp:25]:
```suggestion
	uint32_t duration; // Duration in milliseconds
```

Comment (AJenbo) [reply Source/error.cpp]: Message was unclear, but I think it's the double space

**Pull request merged**